### DISCUSSION TITLE
Setting by long pressing third button - user choose in menu between extended RDS and Instant Sleep Display (as it works in AM mode now)

### DISCUSSION THREAD

**Discussion by leryfm** [PE5PVB/TEF6686_ESP32#464] [Ideas]
Hello! 

Let me make a proposal to implement **setting the screen to sleep by long pressing the button** (this is already done in AM mode now and works exellent and pleasant), is it possible to make a similar function (choice in the menu between the extended RDS long-press displaying and the function to put the screen to sleep by 3rd button)? 

The fact is that with DX-reception, extended RDS is not always needed (because of too weak receiving signals to run RDS or some stations transmit a signal without RDS at all, respectively, there are situations where the extended RDS, called by a long button press, is useless. and it would be more useful to put the screen to sleep by pressing this), but manually putting the screen to sleep would be more convenient than waiting for a configured sleep-timer. 

This is implemented, for example, in mobile phones (I always set the screen timeout for 5 minutes, and put the screen to sleep manually, with the power button when I don’t need it). 
Believe me, this is much more pleasant than when he falls asleep during a timeout.

In AM mode this is exactly how the work is implemented. It would be nice if the **user could also choose** in FM mode - what is more important to him - extended RDS by long pressing the third button, or convenient sleep of the screen, and could switch depending on the reception conditions. I repeat that this is not at all the same as setting a timer for the screen to go to sleep, it would be much more convenient.

I would like this idea to be implemented, is it possible?

**Comment by PE5PVB** [comment]
Yes, it is possible. But what is the gain of turning off the screen? Can you give an example?

**Reply by leryfm** [reply]
For example, sometimes I listen to the radio at night, when the lights off, and when everyone nearby is already asleep.
Having tuned in to the desired frequency, I don’t plan to change it further, because, I liked the song or program and I don’t plan to change the frequency yet, but I will later. 
(in this case, the station could be without RDS, or simply DX-reception, when RDS not decoded, or simply not important at this moment). 
To prevent the light from the display from hitting my eyes in complete darkness, I turn off the screen (now this is conveniently implemented in the AM mode, but not in FM). For FM i have to go into the menu, set a timer for 3 seconds, confirm everything, instead of pleasantly long-pressing one button. This is inconvenient, I would like to improve this process. 
If I need to change the frequency again, I can do it “by touch” (there is a corresponding function "rotate when sleep", thank you for it, by the way! I found this option useful, I have it enabled). The option with a timer, even in a small size of 3 seconds, is inconvenient; I would like to be able to do exactly the same behavior as in AM now. 
Of course, there are users for whom the extended RDS is important, and for them it is necessary to provide a choice in the menu of how the receiver should behave by long pressing the third button - 1) as it is now (extended RDS window шт FM), 2) to put the screen to sleep manually in FM , same as in AM mode.

**Comment by PE5PVB** [comment]
What about holding button for > 4 sec. to activate screen off?

**Comment by leryfm** [comment]
Do you mean that it can be implemented so that a 2-second press, as now, would initiate an extended RDS, but if continue to hold the button pressed and wait a little longer, the screen would switch off? 

It fits, of course, I didn’t suspect it was possible :) This solution seems quite convenient.
